src/NuGet.Core/NuGet.Configuration/Utility/SettingsUtility.cs

```diff
@@ -32,6 +33,17 @@ public static string GetRepositoryPath(ISettings settings)
             return path;
         }
 
+        public static int GetMaxHttpRequest(ISettings settings)
+        {
+            var max = settings.GetValue(ConfigSection, MaxHttpRequest, isPath: false);
+            if (!string.IsNullOrEmpty(max) && int.TryParse(max, out var result))
+            {
+                return result;
+            }
+
+            return 0;
```

Comment:
should the default be 1 instead?

Reply:
no, if this not set, we will not set throttle, unlimited is the default.